comp/metadata/packagesigning/packagesigningimpl/decryptgpg.go

```diff
@@ -0,0 +1,131 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package packagesigningimpl
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	pkgUtils "github.com/DataDog/datadog-agent/comp/metadata/packagesigning/utils"
+	pgp "github.com/ProtonMail/go-crypto/openpgp"
+)
+
+// signingKey represents relevant fields for a package signature key
+type signingKey struct {
+	Fingerprint    string                `json:"signing_key_fingerprint"`
+	ExpirationDate string                `json:"signing_key_expiration_date"`
+	KeyType        string                `json:"signing_key_type"`
+	Repositories   []pkgUtils.Repository `json:"repositories"`
+}
+
+type repoFile struct {
+	filename     string
+	repositories []pkgUtils.Repository
+}
+
+const (
+	noExpDate  = "9999-12-31"
+	formatDate = "2006-01-02"
+)
+
+// decryptGPGFile parse a gpg file (local or http) and extract signing keys information
+// Some files can contain a list of repositories.
+func decryptGPGFile(cacheKeys map[string]signingKey, gpgFile repoFile, keyType string, client *http.Client, logger log.Component) {
+	var reader io.Reader
+	if strings.HasPrefix(gpgFile.filename, "http") {
+		response, err := client.Get(gpgFile.filename)
+		if err != nil {
+			logger.Infof("Error while reading %s: %s", gpgFile.filename, err)
```

Comment:
nit
```suggestion
			logger.Infof("Error while reading GPG file %s: %s", gpgFile.filename, err)
```

Similar comment for other errors in this method